mmv1/templates/terraform/post_update/compute_subnetwork.go.erb

```diff
@@ -0,0 +1,72 @@
+if v, ok := d.GetOk("send_secondary_ip_range_if_empty"); ok && v.(bool) {
+	if sv, ok := d.GetOk("secondary_ip_range"); ok {
+		configValue := d.GetRawConfig().GetAttr("secondary_ip_range").AsValueSlice()
```

Comment:
What case does this handle, if we've presumably set a known zero value for the field in the CustomizeDiff? Theoretically wouldn't we always get the correct value from `d.Get`?

Reply:
This is necessary when we remove ConfigModeAttr. After removing it, d.GetChange/d.HasChange loses the diff from the plan, so this ended up necessary to actually trigger the update. It works the same for with and without ConfigModeAttr

Reply:
> d.GetChange/d.HasChange loses the diff from the plan

Just to make sure I have this right- `terraform plan` displays a diff, but those functions- the "safe" functions in the SDK- report that it did not?

Reply:
yes this is a local run [without ConfigModeAttr](https://github.com/c2thorn/terraform-provider-google-beta/commit/fc830b85261a4e8f4bda19e487edd093b19e39fd#diff-0fb3f296c74e78df5ca33f93d3f3ada9f40d9fad606ebf4207149b1874bc441bR287) and some added debug logs: gpaste/5449355903631360

`setting secondary_ip_range to newly empty` line from the customDiff appears twice as expected

paste/5449355903631360#l=399 are the print logs I added. Here's [the code](https://github.com/c2thorn/terraform-provider-google-beta/commit/fc830b85261a4e8f4bda19e487edd093b19e39fd#diff-0fb3f296c74e78df5ca33f93d3f3ada9f40d9fad606ebf4207149b1874bc441bR1086)

For good measure, I even added another [debug log](https://github.com/c2thorn/terraform-provider-google-beta/commit/fc830b85261a4e8f4bda19e487edd093b19e39fd#diff-0fb3f296c74e78df5ca33f93d3f3ada9f40d9fad606ebf4207149b1874bc441bR1013) for if the previous MMv1-generated update code is even triggering. It does not appear

Reply:
Thanks!